packages/driver/cross-origin-testing.md

```diff
@@ -0,0 +1,141 @@
+# Cross-origin Testing Technical Overview
+
+The goal of this document is to give a technical overview of the architecture behind the **cy.origin()** command, which enables cross-origin testing in Cypress.
+
+## Definitions
+
+See [Node.js’s URL doc](https://nodejs.org/api/url.html#url-strings-and-url-objects) for a handy breakdown of URL parts
+
+- **domain**: The top-level domain or super-domain of a URL. A hostname without the subdomain. (e.g. `example.com`)
+- **origin**: The combination of the protocol, hostname, and port of a URL. For the purposes of Cypress, the subdomain is irrelevant. (e.g. `http://example.com:3500`)
+- **top**: The main window/frame of the browser
+- **primary origin**: The origin that top is on
+- **secondary origin**: Any origin that is not the primary origin
+- **primary driver**: The Cypress driver that run in **top** on the primary origin
+- **secondary driver**: Any Cypress driver that run in a **spec bridge**, interacting with a secondary origin
+
+## Frame architecture (outside of multi-domain)
+
+When testing a single origin, all 3 frames are loaded on that origin.
+
+Components:
+
+```mermaid
+graph TD;
+    top["top frame: domain1.com"]-->specFrame["spec frame: domain1.com"];
+    top-->aut["AUT frame: domain1.com"];
+```
+
+**top** communicates directly and synchronously with the **spec frame** and the **AUT frame**. The **spec frame** runs the spec, which uses the driver to run commands and interact with the **AUT**.
+
+In a single test (`it` + hooks), the **AUT** must remain on the same origin or a cross-origin error will occur, as **top** can no longer interact with the **AUT** in that circumstance. Different tests can visit different origins. In this case, we change **top** to the new origin, which also runs the **spec frame** on that origin. This navigation causes the spec to run again. We skip any already-run tests and resume.
+
+## Frame architecture (with multi-domain)
```

Comment:
```suggestion
## Frame architecture (multi-domain)
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21241/commits/c34e56c4e1b477dfe0cb896e5827d287f819d3c5